src/context.rs

```diff
@@ -403,3 +403,107 @@ impl<A> FusedFuture for ReceiveFuture<A> {
         self.0.is_terminated()
     }
 }
+
+pub struct TickFuture<'a, A> {
+    state: TickState<'a, A>,
+}
+
+#[cfg(feature = "instrumentation")]
+impl<'a, A> TickFuture<'a, A> {
+    /// Return the handler's [`tracing::Span`](https://docs.rs/tracing/latest/tracing/struct.Span.html)
+    /// and the future, if the message is not a shut down. This can be used to log messages into the
+    /// span when required, such as if it is cancelled later due to a timeout.
+    pub fn with_span(self) -> (Option<tracing::Span>, TickFuture<'a, A>) {
```

Comment:
Can we even call this on a pinned future? Once it has been polled, it will be pinned.

Reply:
That's true, I do not believe so - example might make the intended use clearer though so I'll add one